### ISSUE TITLE
Media source input removal causing audio cut off and monitoring desync on stream / recordings

### ISSUE THREAD

**Issue by jdudetv** [obsproject/obs-studio#10907]
### Operating System Info

Windows 10

### OBS Studio Version

30.2.0-beta4

### OBS Studio Log URL

https://obsproject.com/logs/ZWSEUFbSezIGCyNo

### Expected Behavior

Removing a media source once it has finished playing should have the audio output to stream / recording the exact same as what monitoring hears.

### Current Behavior

Audio monitoring and audio mixer show audio playing but the last 0.5-1s of audio is being clipped off the end of the media source even though the broadcaster can still hear it through monitoring and see it in the mixer. If the clip is short enough no audio will output to the stream / recording at all while the broadcaster will hear the audio in monitoring.

### Steps to Reproduce

1. create a media source in OBS
2. put monitor and output on for that source
3. use OBS ws to remove the input on media playback ended event
4. be live or recording too file
5. play the media source and then have it removed right as the playback ends
6. the last bit of the media file will be clipped off and not heard in the recording or broadcast.

**Comment by RytoEX** [comment]
Does this also occur in OBS Studio 30.1.2 or 30.0.2?

**Comment by jdudetv** [comment]
Yes, sorry this was a bug i'd run into before using the beta just happened to be the version I was using at the time when it came up again now.
